jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SizeLimitHandler.java

```diff
@@ -59,8 +59,12 @@ public boolean handle(Request request, Response response, Callback callback) thr
         if (contentLengthField != null)
         {
             long contentLength = contentLengthField.getLongValue();
-            if (contentLength > _requestLimit)
-                throw new BadMessageException(413, "Request body is too large: " + contentLength + ">" + _requestLimit);
+            if (_requestLimit >= 0 && contentLength > _requestLimit)
+            {
+                String s = "Request body is too large: " + contentLength + ">" + _requestLimit;
+                callback.failed(new HttpException.RuntimeException(413, s));
```

Comment:
Sorry, but I've changed my mind again. 

If this is in a context, we want the context error page to be written, so either revert to the throw (which can be caught by a ContextHandler) or call Response.writeError here.

Reply:
reverted to the `Response.writeError`

the throw was causing a failure in this test
https://github.com/GoogleCloudPlatform/appengine-java-standard/blob/3.0.x/runtime/test/src/test/java/com/google/apphosting/runtime/jetty/SizeLimitHandlerTest.java